clients/privacy-center/packages/fides-consent/src/lib/cookie.ts

```diff
@@ -57,32 +129,58 @@ export const makeDefaults = ({
   return defaults;
 };
 
-export const getConsentCookie = (
-  defaults: CookieKeyConsent
-): CookieKeyConsent => {
+export const getOrCreateFidesCookie = (
+  defaults?: CookieKeyConsent
+): FidesCookie => {
+
+  // Create a default cookie and set the configured consent defaults
+  const defaultCookie = createFidesCookie(defaults);
+
   if (typeof document === "undefined") {
-    return defaults;
+    return defaultCookie;
   }
 
-  const cookie = getCookie(CONSENT_COOKIE_NAME, CODEC);
-  if (!cookie) {
-    return defaults;
+  // Check for an existing cookie for this device
+  const cookieString = getCookie(CONSENT_COOKIE_NAME, CODEC);
+  if (!cookieString) {
+    return defaultCookie;
   }
 
   try {
-    const parsed: CookieKeyConsent = JSON.parse(cookie);
-    return {
+    // Parse the cookie and check it's format; if it's structured like we
+    // expect, cast it directly. Otherwise, assume it's a previous version of
+    // the cookie, which was strictly the consent key/value preferences
+    let parsedCookie: FidesCookie;
+    const parsedJson = JSON.parse(cookieString);
+    if ("consent" in parsedJson && "fides_meta" in parsedJson) {
+      // Matches the expected format, so we can use it as-is
+      parsedCookie = parsedJson;
+    } else {
+      // Missing the expected format, so we parse it as strictly consent
+      // preferences and "wrap" it with the default cookie style
+      parsedCookie = {
+        ...defaultCookie,
+        consent: parsedJson,
```

Comment:
This gives us backwards-compatibility with an unwrapped cookie.

Reply:
I tested this manually, but an automated test would be better.

Reply:
agreed, it'd be great to have a test which has the old format as well as the new one. would be especially useful for if we continue to add attributes to the cookie and need it still to be backwards compatible. that said... I don't think `fides-consent` as it exists right now has testing configured 😿 but hopefully once our turborepo env is set up, we'll be able to test this more easily!

Reply:
😉 2 days ago I had the same idea: https://github.com/ethyca/fides/pull/3105/commits/18eaa632000d160c843e83a3512b5affb7d4198e